api/segments/models.py

```diff
@@ -181,9 +183,15 @@ def does_identity_match(
 
         for trait in traits:
             if trait.trait_key == self.property:
-                if trait.value_type == INTEGER:
+                if trait.value_type in [INTEGER, FLOAT] and self.operator == MODULO:
```

Comment:
I believe that, due to the way in which traits are deserialized, the value_type will always be `STRING` since `a|b` will always be read as a string. We should simply pass the `trait.trait_value` or `trait.string_value` to the `_check_modulo_operator` method.

Reply:
Correct me if I'm wrong - `trait.value_type` is the data type of the trait belonging to `Identity`, which indeed can be Integer or Float or String. The modulo operator will be applicable only for the traits that are Integer or Float (example, user-ids, count of something, etc) and hence the condition is added.
Here `self.value` i.e. the `value` attribute of `Condition` object will always be `String` since it'll be `a|b` format.

Reply:
I have tested modulo evaluation for Integer, Float and String trait. It get's evaluated for Int and Float and skips the check for String.

Reply:
Sorry, yes, you are indeed correct! That's my mistake. Thanks for clarifying.